MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.Services.Tests/Services/ListCacheTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyPerfectOnboarding.Contracts.Database;
+using MyPerfectOnboarding.Contracts.Models;
+using MyPerfectOnboarding.Services.Services;
+using MyPerfectOnboarding.Tests.Utils.Builders;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace MyPerfectOnboarding.Services.Tests.Services
+{
+    [TestFixture]
+    internal class ListCacheTests
+    {
+        private readonly ListItem[] _items =
+        {
+            ListItemBuilder.CreateItem("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC", "aaaaa", "1589-12-03", "1896-04-07"),
+            ListItemBuilder.CreateItem("11AC59B7-9517-4EDD-9DDD-EB418A7C1644", "dfads", "4568-06-23", "8569-08-24")
+        };
+
+        private ListCache _listCache;
+        private IListRepository _listRepository;
+
+        [SetUp]
+        public void Init()
+        {
+            _listRepository = Substitute.For<IListRepository>();
+
+            _listCache = new ListCache(_listRepository);
+
+            _listRepository.GetAllItemsAsync().Returns(_items);
+        }
+
+        [Test]
+        public async Task AddItemAsync_item_AddItemIntoRepository()
+        {
+            var stringId = "22AC59B7-9517-4EDD-9DDD-EB418A7C1678";
+            var newItem = ListItemBuilder.CreateItem(stringId, "newItem");
+            var id = Guid.Parse(stringId);
+            _listRepository.AddItemAsync(newItem).Returns(newItem);
+
+            var addedItem = await _listCache.AddItemAsync(newItem);
+
+            Assert.Multiple(async () => {
+                await _listRepository.Received(1).GetAllItemsAsync();
+                await _listRepository.Received(1).AddItemAsync(newItem);
+                Assert.That(_listCache.Items, Contains.Key(id));
+                Assert.That(addedItem, Is.EqualTo(newItem));
+            });
+        }
+
+        [Test]
+        public async Task DeleteItemAsync_itemId_DeleteItemFromRepository()
+        {
+            var id = _items[0].Id;
+            _listRepository.DeleteItemAsync(id).Returns(_items[0]);
+
+            var deletedItem = await _listCache.DeleteItemAsync(id);
+
+            Assert.Multiple(async () => {
+                await _listRepository.Received(1).GetAllItemsAsync();
+                await _listRepository.Received(1).DeleteItemAsync(id);
+                Assert.That(_listCache.Items.Keys, Is.Not.Contains(id));
+                Assert.That(deletedItem, Is.Not.Null);
+                Assert.That(deletedItem.Id, Is.EqualTo(id));
```

Comment:
I was thinking that these 2 assertions deserve just one. Here are two options crossing my mind:
```suggestion
                Assert.That(deletedItem, Is.Not.Null.And.Property(nameof(IListItem.Id)).EqualTo(id));
```
or
```suggestion
                Assert.That(deletedItem?.Id, Is.EqualTo(id));
```
What do you think?